150 death animations

MrJackey opened the pull request:
## Summary
Fixes 150
Added deathscreen which appears on death.
It teleports you back to checkpoint midanimation if there is one, otherwise reloads scene.
Reloading scene looks a bit weird but wasn't sure how else to do it.
I'm not proud of the animator but it works.

Commit:
Created death screen animations

Commit:
Connected death screen to player death

Commit:
Dies differently depending on source

Commit:
Added human piercing and cleaned up helper

Commit:
Updated piercing animation spritesheets

smeas commented on Assets/Scripts/Common/Health.cs:
```diff
@@ -43,18 +43,17 @@ public class Health : MonoBehaviour {
 
-
-	public void TakeDamage(float amount) {
+	public void TakeDamage(float amount, DamageType damageType = DamageType.Regular) {
```
Now you can't invoke this method from a UnityEvent in the editor. You shoud make it an overload instead.

smeas commented on Assets/Scripts/Common/Health.cs:
```diff
@@ -84,4 +83,10 @@ public class Health : MonoBehaviour {
\ No newline at end of file
+
+
+	public enum DamageType {
+		Regular,
+		Piercing,
+	}
```
I don't think this enum should be nested inside the `Health` class.

smeas replied:
The damage types are very generic while the death animations are not. What if you get pierced by something that is not a spike?
Maybe we should have something like this instead:
```cs
public enum DamageSource {
	Generic,
	Spikes,
}
```
?

Addehman reviewed (approved):
Looks good to me! 👌

smeas commented on Assets/Scripts/UI/DeathScreenHelper.cs:
```diff
@@ -0,0 +1,93 @@
+	}
+
+	private void Start() {
+		checkpointManager = FindObjectOfType<CheckpointManager>();
```
Maybe add a tag to the checkpoint manager and use that to find it instead to avoid having to search through the whole scene for one.

smeas commented on Assets/Scripts/UI/DeathScreenHelper.cs:
```diff
@@ -0,0 +1,93 @@
+	}
+
+	private IEnumerator CoSelectBodyAnimation(Health.DamageType damageType) {
+		while (!bodyAnimator.isInitialized)
+			yield return null;
```
I've never used animation events before but maybe theyre a good fit for this? That way the animation can invoke a function to start the body animation.

Commit:
Made suggested changes

Extracted prev damageType -> damageSource out of Health class.
Added checkpointmanager tag and animation event for body animation.

smeas commented on Assets/Scripts/UI/DeathScreenHelper.cs:
```diff
@@ -35,7 +36,9 @@ public class DeathScreenHelper : MonoBehaviour {
 
 	private void Start() {
-		checkpointManager = FindObjectOfType<CheckpointManager>();
+		GameObject go = GameObject.FindWithTag("CheckpointManager");
+			if (go != null)
+				checkpointManager = go.GetComponent<CheckpointManager>();
```
Indentation
```suggestion
		GameObject go = GameObject.FindWithTag("CheckpointManager");
		if (go != null)
			checkpointManager = go.GetComponent<CheckpointManager>();
```

smeas reviewed (changes requested):
The checkpoint manager in the scene is missing its tag. Maybe add a prefab for it and replace it in the scene? (merge master first!)

Commit:
Merge branch 'master' into feature/150-death-animations

Commit:
Added checkpoints prefab and updated tutorial scene

MrJackey committed:
Fixed indentation

Co-authored-by: Jonatan Johansson <smeas@users.noreply.github.com>

smeas reviewed (approved):
:shipit:
